### DISCUSSION TITLE
Is sea-orm considered stable?

### DISCUSSION THREAD

**Discussion by 7flying** [SeaQL/sea-orm#1619] [Q&A] [answered]
My team and I are in the process of selecting an ORM and sea-orm might be "the one", but the last release is `0.11.3` which shouldn't be considered a stable release if we follow the standard conventions. 

So, just double-checking, is sea-orm considered stable?

Thanks!

**Comment by tyt2y3** [comment, marked as answer]
Hi, thank you for the prompt! Here is our maintenance principles:

1. Release cycle. We used to release a major version every month, but it becomes roughly every 3 months these days. Each major release will come with many new features, along with some breaking changes. We try to document all breaking changes with upgrade instructions. Most of the time a global "find and replace" is all you need.

2. Bug fixes. We will release minor versions as needed to mitigate bugs.

3. Stability. A major bump is unavoidable when we want to upgrade a dependency (i.e. SQLx). Both SQLx and SeaORM are used in production by many projects (look at https://github.com/SeaQL/sea-orm/blob/master/COMMUNITY.md). SQLx is pretty stable now in terms of API and features. SeaORM is still evolving, but it is a good thing, because it brings new features and enhancements. We have a large API surface, and we have a comprehensive CI to safeguard that. We evaluate every breaking change very carefully.

Final note: I am considering to promote the next release `0.12` to be a LTS version.

**Reply by 7flying** [reply]
Thank you for the detailed response :+1:
